message_coding/templates/_layouts/form_base.html

```diff
@@ -0,0 +1,41 @@
+{% load widget_tweaks %}
+
+        {# borrowed from http://wisercoder.com/django-forms-bootstrap/ #}
+        <form class="form" action="" method="POST" {% if form.is_multipart %}enctype="multipart/form-data"{% endif %}>
+            {% csrf_token %}
+            {{ form.media }}
+            {% if form.errors %}
+                <div class="alert alert-error">
+                    <ul>
+                        {% for error in form.non_field_errors %}
+                            <li>{{ error }}</li>
+                        {% endfor %}
+                    </ul>
+                </div>
+            {% endif %}
+            {% for field in form.visible_fields %}
+                <div class="form-group {{ field.html_name }} {% if field.errors %}error{% endif %}">
```

Comment:
I think this should use `has-error` class rather than `error` ([details](http://getbootstrap.com/css/#forms-control-validation)).